### PR TITLE
updated display conditions for upgrade notice

### PR DESCRIPTION
**Pull request by akshayurankar48**
### Description
[BSF-PR-SUMMARY]

### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: src/Components/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -1,14 +1,71 @@
...
+            return dismissed !== 'true';
+        } catch (error) {
+            console.error('Dashboard - localStorage check error:', error);
+            return true; // Default to showing the notice
+        }
+    });
+
+    // Function to handle closing the upgrade notice
+    const handleCloseUpgradeNotice = async () => {
+        console.log('Dashboard - Closing upgrade notice'); // Debug log
```
Comment (sushmak02) [inline src/Components/Dashboard/Dashboard.jsx:32]: remove unwanted console logs

File: admin/class-hfe-addons-actions.php [hunk 2]
```diff
@@ -572,6 +574,23 @@ public function save_analytics_option() {
...
+		 * Dismiss upgrade notice
+		 *
+		 * @return void
+		 */
+		public function dismiss_upgrade_notice() {
+			// Verify nonce for security
+			if ( ! isset( $_POST['nonce'] ) || ! wp_verify_nonce( $_POST['nonce'], 'hfe_dismiss_upgrade_notice' ) ) {
+				wp_send_json_error( __( 'Security check failed.', 'header-footer-elementor' ) );
+				return;
+			}
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply admin/class-hfe-addons-actions.php:593]: updated @sushmak02

File: admin/class-hfe-addons-actions.php [hunk 3]
```diff
@@ -572,6 +574,29 @@ public function save_analytics_option() {
...
+		/**
+		 * Dismiss upgrade notice
+		 *
+		 * @return void
+		 */
+		public function dismiss_upgrade_notice() {
+
+			// Check if user has permission to manage options.
+			if ( ! current_user_can( 'manage_options' ) ) {
+				wp_send_json_error( __( 'You do not have permission to perform this action.', 'header-footer-elementor' ) );
```
Comment (sushmak02) [inline admin/class-hfe-addons-actions.php:586]: return;

File: src/App.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import "@fontsource/figtree/400.css"; // Specify weight
 import "@fontsource/figtree/400-italic.css"; // Specify weight and style
 import CustomRouter from "router/customRouter";
 import { Loader } from "@bsf/force-ui";
-import UpgradeNotice from "@components/UpgradeNotice";
+// import UpgradeNotice from "@components/UpgradeNotice";
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply src/App.js:7]: done @ShubhamGupta05

File: admin/class-hfe-addons-actions.php (hunk 3 above)
Comment (akshayurankar48) [reply admin/class-hfe-addons-actions.php:586]: done @sushmak02

File: admin/class-hfe-addons-actions.php [hunk 5]
```diff
@@ -572,6 +574,29 @@ public function save_analytics_option() {
...
+			}
+
+			// Verify nonce for security
+			if ( ! isset( $_POST['nonce'] ) || ! wp_verify_nonce( $_POST['nonce'], 'hfe_dismiss_upgrade_notice' ) ) {
+				wp_send_json_error( __( 'Security check failed.', 'header-footer-elementor' ) );
+				return;
+			}
+
+			// Update option to remember the dismissal
+			update_option( 'hfe_upgrade_notice_dismissed', 'true' );
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply admin/class-hfe-addons-actions.php:596]: Tried with meta and local storage its not working , but its working perfectly with option_name @sushmak02

File: src/Components/Compare/Upgrade.jsx [hunk 6]
```diff
@@ -3,15 +3,74 @@ import ExtendWebsite from "@components/Dashboard/ExtendWebsite";
...
+    // Check if upgrade notice was dismissed (this will be handled by PHP)
+    const [showTopBar, setShowTopBar] = useState(() => {
+        // First check server-side dismissal
+        if (window.hfe_admin_data && window.hfe_admin_data.upgrade_notice_dismissed) {
+            return false;
+        }
+        
+        // Fallback to localStorage check
+        try {
+            const dismissed = localStorage.getItem('uae_upgrade_notice_dismissed');
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply src/Components/Compare/Upgrade.jsx]: updated @sushmak02

File: src/Components/Widgets/Features.jsx [hunk 7]
```diff
@@ -1,16 +1,76 @@
...
+            }
+        } catch (error) {
+            console.error('Features - AJAX error:', error);
+        }
+    };
+
     return (
         <>
+         {showTopBar && (
+                <UpgradeNotice onClose={handleCloseUpgradeNotice} />
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply src/Components/Widgets/Features.jsx:52]: done @sushmak02

File: admin/class-hfe-addons-actions.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (ShubhamGupta05) [reply admin/class-hfe-addons-actions.php:597]: @akshayurankar48 We have to make it for user meta it will work.  Instead of saving it like 
			update_option( 'hfe_upgrade_notice_dismissed', 'true' );
			make it like  this one 
update_user_meta( get_current_user_id(), 'hfe_upgrade_notice_dismissed', 'true' ); 

and to  fetch use this

 get_user_meta( get_current_user_id(), 'hfe_upgrade_notice_dismissed', true )

File: inc/class-hfe-settings-page.php [hunk 8]
```diff
@@ -462,6 +463,8 @@ public function enqueue_admin_scripts() {
 			'ajax_url'              => admin_url( 'admin-ajax.php' ),
 			'nonce'                 => wp_create_nonce( 'hfe-admin-nonce' ),
 			'installer_nonce'       => wp_create_nonce( 'updates' ),
+			'upgrade_notice_dismissed' => $upgrade_notice_dismissed,
+			'upgrade_notice_nonce'  => wp_create_nonce( 'hfe_dismiss_upgrade_notice' ),
```
Comment (ShubhamGupta05) [inline inc/class-hfe-settings-page.php:467]: @akshayurankar48  We can use the old nonce only; we don't need to add and pass a new nonce.

Comment (akshayurankar48) [reply inc/class-hfe-settings-page.php:467]: done @ShubhamGupta05

File: inc/class-hfe-settings-page.php [hunk 9]
```diff
@@ -443,6 +443,7 @@ public function enqueue_admin_scripts() {
 		wp_enqueue_script( 'hfe-admin-script', HFE_URL . 'admin/assets/js/ehf-admin.js', [ 'jquery', 'updates' ], HFE_VER, true );
 	
 		$is_dismissed = get_user_meta( get_current_user_id(), 'hfe-popup' );
+		$upgrade_notice_dismissed = get_option( 'hfe_upgrade_notice_dismissed', 'false' ) === 'true';
```
[the remark this answers is not in the document]
Comment (akshayurankar48) [reply inc/class-hfe-settings-page.php]: updated @ShubhamGupta05

**Review by sushmak02 — APPROVED** [review]
